Gems/Atom/Tools/AtomToolsFramework/Code/Source/Viewport/ModularViewportCameraController.cpp

```diff
@@ -323,11 +323,11 @@ namespace AtomToolsFramework
     void PlaceholderModularCameraViewportContextImpl::SetCameraTransform(const AZ::Transform& transform)
     {
         m_cameraTransform = transform;
-        m_viewMatrixChangedEvent.Signal(AzFramework::CameraViewFromCameraTransform(Matrix4x4FromTransform(transform)));
+        m_viewMatrixChangedEvent.Signal(
+            AZ::Matrix4x4::CreateFromMatrix3x4(AzFramework::CameraViewFromCameraTransform(AZ::Matrix3x4::CreateFromTransform(transform))));
```

Comment:
In the description it's said that using `Matrix3x4` instead of `Matrix4x4` "makes a lot more sense", can you elaborate why that's the case? What's the gain with this change?

When it's signaling the matrix change it's doing now more transformations than before... I'm wondering what is the overall gain which in return justifies this additional compute.

Thanks

Reply:
The gain is because a `Matrix4x4` can represent more than just an 'affine' transformation that is all I want to represent with a camera view/transform. Limiting what is accepted in this part of the API makes it clearer what type of transform we want. There are more conversions that need to happen in other places if we leave the type as `Matrix4x4`. The reason `View.cpp` uses `Matrix4x4` throughout is to interop more closely with the `float4x4`s in the shaders.

I do think this is the right decision as the type is a closer 'fit' to what should be passed in. If there was ever a performance problem in future we could always create an overload but I think that is pretty unlikely

Reply:
`Matrix3x4` can still hold any numbers representing anything in it. It's just missing the last column `[0,0,0,1]`, but you can still represent "any" matrix with it, it doesn't have to be a affine transformation. If you really want to limit it to an affine transformation maybe it could use AZ::Transform directly?
I suppose one gain is the memory saving (4x4floats vs 3x4floats).
Anyway, this is more a discussion rather than PR feedback at this point. Nothing wrong with the code, so I'll approve it.